.github/workflows/prereq.yml

```diff
@@ -18,13 +20,24 @@ jobs:
           fetch-depth: 500
       - name: Environment
         run: |
-          echo $GITHUB_REF_NAME
-          git rev-list -1 HEAD
-          git rev-list -1 HEAD^1
-          git rev-list -1 HEAD^2
+          echo "GITHUB_REF_NAME: $GITHUB_REF_NAME"
+          echo "GITHUB_BASE_REF: $GITHUB_BASE_REF"
+          echo "HEAD: $(git rev-list -1 HEAD)"
+          if test -n "$GITHUB_BASE_REF"; then
+            BASEB=$(git rev-list -1 HEAD^1)
+            HEADB=$(git rev-list -1 HEAD^2)
+          else
+            git fetch --no-tags --no-recurse-submodules --depth=500 origin main
+            BASEB=origin/main
+            HEADB=HEAD
+          fi
+          echo "BASEB: $BASEB"
+          echo "HEADB: $HEADB"
```

Comment:
these two lines look redundant with the following two.. You'll see the values in the logs, would you not

Reply:
The following two lines depict the official way of env vars passing to the next steps of a job: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable.
Probably, it was meant that reporting the values in the Environment step could be omitted due to they can be spotted in the commands of the next steps. If we prefer such way, then yes, the lines 34-35 can be omitted.